client/scripts/views/pages/community_cards.tsx

```diff
@@ -78,62 +78,33 @@ class ChainCard implements m.ClassComponent<ChainCardAttrs> {
   }
 }
 
-type CommunityCardAttrs = { community: ChainInfo };
-
-class CommunityCard implements m.ClassComponent<CommunityCardAttrs> {
-  view(vnode) {
-    const { community } = vnode.attrs;
-
-    const redirectFunction = (e) => {
-      e.preventDefault();
-      localStorage['home-scrollY'] = window.scrollY;
-      m.route.set(`/${community.id}`);
-    };
+// Corresponds to 'ChainCategoryTypes' id numbering (which currently only includes DeFi and DAO)
+const filterMap = {
+  DeFi: 1,
+  DAO: 2,
+  ERC20: 3,
+  Cosmos: 4,
+  Substrate: 5,
+  Ethereum: 6,
+};
 
-    let prettyDescription = '';
-    if (community.description) {
-      prettyDescription =
-        community.description[community.description.length - 1] === '.'
-          ? community.description
-          : `${community.description}.`;
+class HomepageCommunityCards implements m.ClassComponent {
+  private filters: Array<boolean>;
+  private categoryMap: { [chain: string]: number[] };
```

Comment:
there's a solution here where we can store all of the ChainBases, ChainCategoryTypes, and ChainNetworks in a map of `string => boolean` that can be updated directly. for filtering, we can just iterate through the three arrays above, checking the map `if true, then filter`.